### ISSUE TITLE
Repeater field - limited use

### ISSUE THREAD

**Issue by mary151** [CrocoBlock/suggestions#4235]
Hello,

I think all users would like more features for repeater fields.

For example: 
- Add background property (image, color, slideshow etc.) from repeater field value.
- Add css id/class property in listing sections from repeater filed value. Or dynamic css id values for dynamic repeater itself. For ex. If I have repeater with field "css-id", to be able to set it as css id in main section settings. In this way I may create internal page navigation sections dynamically: #section1, #section2 etc.
- To be able to create whole tooltip with image/ icon/ text from repeater. This will be very usefull in woocommerce product page to add icons with tooltips.

If someone has other ideas, please write them below.

**Comment by rtpHarry** [comment]
Yeah this is very disappointing that the repeater barely works, except for in a few controls that have support specifically added to it.

I want to be able to set dynamic visibility based on a key being empty or not.

I also want to be able to wire repeater values in via Dynamic Tag's so I can pass the settings through to my elements. 

At the moment it seems that neither of these scenarios are available.

You can just use it with the dynamic field element which is fine if you just have a simple img or table row to generate but no good for more complex scenarios.

**Comment by rtpHarry** [comment]
I have implemented a dynamic tag which can be used to bind values to elements within JetEngine listing templates:

https://github.com/rtpHarry/jet-engine-repeater-values

This will solve a number of the issues by making the data available to a lot of the fields that don't have explicit repeater support.

You can use the plugin the same as any other dynamic tag, it's listed under a RunThings.dev category:

![image](https://user-images.githubusercontent.com/1038062/145736158-e26901c2-f0c2-4fc3-83fa-05fc4e1c61f8.png)

And lets you enter in the key, in the same way the dynamic field supports:

![image](https://user-images.githubusercontent.com/1038062/145736223-c77f270e-85ff-47e9-8ba3-79cbd758caec.png)

![image](https://user-images.githubusercontent.com/1038062/145736291-d37377bd-741f-4962-bbf3-de48f09dadd2.png)

This is going to solve my immediate project need as I just wanted to be able to pass values through to a widget I had made in Unlimited Elements.

Hopefully it is useful to others!

**Comment by tealou** [comment]
This is fantastic, Matthew... thank you so much for sharing it back.

**Comment by rtpHarry** [comment]
I was just trying to get my use case working last night, which was for binding a text field.

I have done a little bit more research and it seems there will need to be some further modifications to support the other field types, like images, urls, numbers, colors, etc.

I've put this on the todo list, but if anyone has any specific issues that should be prioritised, please let me know / open them as issues on that repo.

**Comment by rtpHarry** [comment]
I've updated this to v1.1.0 now which adds support for all the types of dynamic tags including text, url, color, post meta, number, gallery, image, and media:

https://github.com/rtpHarry/jet-engine-repeater-values

**Comment by MolyMalon** [comment]
Totally agree!! I'm solving many fields with ACF even though I'm looking forward to doing it with Jet fields.
It would be great to be able to add repeatable fields inside other repeatable fields.

When in business projects the Companies can give you the option of being able to add different addresses but we cannot give the possibility of being able to add different telephone numbers and emails within these addresses since we do not have this possibility.

It would be great, as the friends says, to be able to limit how many repeater fields the user can add.

I still have to use ACF :( with its group of fields and repeatable fields,,,,,,, we need this versatility with Crocoblock fields,,, in the backend!!!

Many thanks..

**Comment by serialtasted** [comment]
@MolyMalon I just opened a new suggestion that you also want.
https://github.com/Crocoblock/suggestions/issues/4972
If you could go there and help me out with a few more suggestions that you see fit would be great
